### PR TITLE
Add Haunt for Scheme as a provider

### PR DESCRIPTION
**Pull request by Siarune**
Creates and registers a build provider for Scheme with the static builder Haunt.
Adds a very basic example to test provider.
Adds docs

### CODE REVIEW

File: docs/pages/docs/providers/scheme.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+ ---
+ title: Scheme
+ ---
+
+ # {% $markdoc.frontmatter.title %}
+
+ Scheme is detected is there is a `haunt.scm` file found.
```
Comment (coffee-cup) [inline docs/pages/docs/providers/scheme.md:7]: Can you add a link to the Haunt homepage.

File: tests/docker_run_tests.rs [hunk 2]
```diff
@@ -1480,3 +1480,12 @@ async fn test_config_toml_file() {
...
     assert!(output.contains("hey there"));
 }
+
+#[tokio::test]
+async fn test_scheme() {
+    let name = simple_build("./examples/scheme").await.unwrap();
+    let output = run_image(&name, None).await;
+    assert!(output.contains("serving site on port 8080"));
+    // let name = simple_build("./examples/scheme").await.unwrap();
+    // assert!(run_image(&name, None).await.contains("serving site on port 8080"));
```
Comment (coffee-cup) [inline tests/docker_run_tests.rs:1490]: 🔫

File: docs/pages/docs/providers/scheme.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+ ---
```
Comment (coffee-cup) [inline docs/pages/docs/providers/scheme.md:1]: Can you also update the `sidebar.ts` file so this is included

**Review by coffee-cup — APPROVED** [review]
Looks awesome! Just a few minor comments

**Comment by Siarune** [comment]
My build plan commands seem out of order, and I can't tell why.

This is my plan definition:
```
let setup = Phase::setup(Some(vec![Pkg::new("haunt")]));
let build = Phase::build(Some("haunt build".to_string()));
let start = StartPhase::new("haunt serve".to_string());

let plan = BuildPlan::new(&vec![setup, build], Some(start));
Ok(Some(plan))
```

So this is what I'd expect the order to be:
```
╔════ Nixpacks v1.29.0 ════╗
║ setup      │ haunt       ║
║──────────────────────────║
║ build      │ haunt build ║
║──────────────────────────║
║ start      │ haunt serve ║
╚══════════════════════════╝
```

But this is what it actually is

```
╔════ Nixpacks v1.29.0 ════╗
║ build      │ haunt build ║
║──────────────────────────║
║ setup      │ haunt       ║
║──────────────────────────║
║ start      │ haunt serve ║
╚══════════════════════════╝
```

**Comment by coffee-cup** [comment]
You can specify phase dependencies. By default the build phase depends on install you might need to specify the dependency manually.

```rust
build.depends_on_phase("setup")
```

**Pull request merged**